src/main/java/htsjdk/samtools/cram/build/SliceFactory.java

```diff
@@ -136,7 +136,13 @@ public List<Slice> createSlices(
                     containerByteOffset,
                     sliceStagingEntry.getGlobalRecordCounter()
             );
-            slice.setReferenceMD5(cramReferenceRegion.getCurrentReferenceBases());
+            if (slice.getAlignmentContext().getReferenceContext().isMappedSingleRef()) {
+                cramReferenceRegion.getReferenceBasesByRegion(
```

Comment:
It's a bit weird to use the getter to update the internal state in order to make the call to "getCurrentBases" work.  As far as I see it's actually only ever used for it's side effect and the return value is always ignored except in tests.  Should the api change to make it clear that that's the primary use?  Should the uses change to use the return value?  Unclear... Maybe add a default method `setCurrentRegion()` that just calls get and throws away the result, just to make it clear that this is not a no-op.

Reply:
Yeah its a little wonky, and its true there are no longer any (non-test) call sites that use the return value directly. I updated the methods to have a `void` return type and renamed them with a `fetch` prefix instead of a get prefix to reinforce their impure character. Let me know if you think thats sufficient.

Reply:
That sounds good to me.